### PR TITLE
Bring back `ml_classical_shadows` demo

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
This reverts the deletion of "Machine learning for quantum many-body problems" tutorial as a non-executable demo compatible with the latest version of PennyLane. 

[sc-68239]

### CODE REVIEW

**Review by KetpuntoG — APPROVED** [review]
Looks good :)

I was just wondering why it is non-executable. In colab everything worked correctly

**Comment by obliviateandsurrender** [comment]
@KetpuntoG our tensorflow version requirements are incompatible with the version of `neural_tangents` that supports the version of `jax` as we do.

File: demonstrations/ml_classical_shadows.metadata.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        "Quantum Machine Learning"
+    ],
+    "tags": [],
+    "previewImages": [
+        {
+            "type": "thumbnail",
+            "uri": "_static/demo_thumbnails/regular_demo_thumbnails/thumbnail_machine_learning_quantum_manybody_problems.png"
+        }
+    ],
+    "seoDescription": "Machine learning for many-body problems",
```
Comment (ikurecic) [inline demonstrations/ml_classical_shadows.metadata.json:20]:
```suggestion
    "seoDescription": "Learn how to apply machine learning to quantum many-body problems by using classical shadow formalism and ML to predict the ground-state properties of the 2D antiferromagnetic Heisenberg model.",
```

File: demonstrations/ml_classical_shadows.py [hunk 2]
```diff
@@ -0,0 +1,862 @@
...
+calculate its ground-state properties, and compute its classical shadow. Finally, we demonstrate
+how to use :doc:`kernel-based learning models <tutorial_kernels_module>` to predict ground-state
+properties from the learned classical shadows. So let's get started!
+
+.. note::
+    This demo is compatible with the latest version of PennyLane and ``neural-tangents==0.6.5``.
+    The latter is required for building the kernel for the infinite network used in training.
+
+
+Building the 2D Heisenberg Model
```
Comment (ikurecic) [inline demonstrations/ml_classical_shadows.py:34]:
```suggestion
Building the 2D Heisenberg model
```

File: demonstrations/ml_classical_shadows.py [hunk 3]
```diff
@@ -0,0 +1,862 @@
...
+
+##############################################################################
+# .. figure:: /_static/demonstration_assets/ml_classical_shadows/exact_correlation_matrix.png
+#    :width: 55%
+#    :align: center
+#    :alt: Exact correlation matrix
+
+
+######################################################################
+# Constructing Classical Shadows
```
Comment (ikurecic) [inline demonstrations/ml_classical_shadows.py:245]:
```suggestion
# Constructing classical shadows
```

File: demonstrations/ml_classical_shadows.py [hunk 4]
```diff
@@ -0,0 +1,862 @@
...
+######################################################################
+# .. figure::  /_static/demonstration_assets/ml_classical_shadows/error_correlation_matrix.png
+#    :align: center
+#    :width: 55 %
+#    :alt: Error in estimating correlation matrix
+#
+
+
+######################################################################
+# Training Classical Machine Learning Models
```
Comment (ikurecic) [inline demonstrations/ml_classical_shadows.py:489]:
```suggestion
# Training classical machine learning models
```

**Review by ikurecic — COMMENTED** [review]
Thanks, @obliviateandsurrender . I left a couple of suggestions to improve this.
If you could address them, that would be good.

I would like to wait to add the large ond OG thumbnails to this pull request, and once that's ready you can leave the merging to me.

File: demonstrations/ml_classical_shadows.metadata.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+    "title": "Machine learning for quantum many-body problems",
+    "authors": [
+        {
+            "id": "whatsis"
```
Comment (ikurecic) [inline demonstrations/ml_classical_shadows.metadata.json:5]:
```suggestion
            "username": "whatsis"
```

**Comment by obliviateandsurrender** [comment]
Thanks @ikurecic, I applied your suggestions! Let me know if there's anything more to be done.

**Review by ikurecic — APPROVED** [review]
Added new images.

**Pull request merged**